internal/admission/validator.go

```diff
@@ -520,6 +523,30 @@ func (noOpRoutesValidator) Validate(_ context.Context, _ *kong.Route) (bool, str
 	return true, "", nil
 }
 
+func (validator KongHTTPValidator) ValidateVault(ctx context.Context, k8sKongVault kongv1alpha1.KongVault) (bool, string, error) {
+	// Ignore KongVaults that are being managed by another controller.
+	if !validator.ingressClassMatcher(&k8sKongVault.ObjectMeta, annotations.IngressClassKey, annotations.ExactClassMatch) {
+		return true, "", nil
+	}
+	config, err := kongstate.RawConfigToConfiguration(k8sKongVault.Spec.Config.Raw)
+	if err != nil {
+		return false, fmt.Sprintf(ErrTextVaultConfigUnmarshalFailed, err), nil
+	}
+	kongVault := kong.Vault{
+		Name:        kong.String(k8sKongVault.Spec.Backend),
+		Prefix:      kong.String(k8sKongVault.Spec.Prefix),
+		Description: kong.String(k8sKongVault.Spec.Description),
+		Config:      config,
+	}
+	// TODO: /schemas/vaults/test does not check "unique" restraint on `prefix` field.
+	// We may need implement the uniqueness check of `prefix`.
```

Comment:
One possible way is to list all `KongVault`s in the store of the controller, which stores all "managed" vaults and then check if the `prefix` is the same with any existing one. Although this may have the following problems:
- Cannot be aware of vaults in other workspaces. I have discussed this problem with @mflendrich and He says it is not a significant problem because there would not be so many vaults.
- Cache sync problem. If a `KongVault` is created but not synced to the controller, we can not know its existence.